Tests/FoundationInternationalizationTests/CalendarTests.swift

```diff
@@ -1220,7 +1220,34 @@ final class CalendarTests : XCTestCase {
         }
 
     }
+
+    func test_addDateOverflow() throws {
+
+        do {
+            let date = Date(timeIntervalSinceReferenceDate: 964779243.351134)
+            let calendar = Calendar(identifier: .gregorian)
+            let components = DateComponents(year: 788960010015224562)
+            let added = calendar.date(byAdding: components, to: date)
+            XCTAssertNil(added)
+        }
+        
+        do {
+            let date = Date(timeIntervalSinceReferenceDate: 849248301.169329)
+            let calendar = Calendar(identifier: .gregorian)
+            let components = DateComponents(year: 9223372036854775556)
+            let added = calendar.date(byAdding: components, to: date)
+            XCTAssertNil(added)
+        }
+    }
+
+    func test_dateComponentsFromDateOverflow() {
+        let calendar = Calendar(identifier: .gregorian)
+        let dc = calendar.dateComponents([.year], from: Date(timeIntervalSinceReferenceDate: Double(Int64.max)))
+
```

Comment:
Just confirming - was there anything else that needed to be here or is this just testing that this doesn't crash?

Reply:
Thanks. Updated